File: solution/1481. Least Number of Unique Integers after K Removals/1481_02.cpp
class Solution {
public:
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        unordered_map<int, int> m;
        priority_queue<int, vector<int>, greater<>> pq;
        
        for (int x : arr)
            ++m[x];
        
        for (auto it : m)
            pq.push(it.second);
        
        while (k > 0) {
            if (k < pq.top())
                break;
            k -= pq.top(); pq.pop();
        }
        return pq.size();
    }
};

File: solution/1481. Least Number of Unique Integers after K Removals/1481_02.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        hp = list(collections.Counter(arr).values())
        heapq.heapify(hp)
        while k > 0:
            if k < hp[0]:
                break
            k -= heapq.heappop(hp)
        return len(hp)
    
